File: backend/app/schemas/audit.py
from typing import Optional, Dict, Any
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    session_id: Optional[str] = None
    actor: str
    action: str
    is_irreversible: bool
    status: str
    details: str
    payload: Optional[Dict[str, Any]] = None
    timestamp: datetime

class AuditStats(BaseModel):
    total_actions: int
    reversible_count: int
    irreversible_count: int
    human_interventions: int
    blocked_actions: int
    compliance_score_pct: float

File: frontend/src/types/audit.ts
export interface AuditLog {
  id: number;
  session_id?: string;
  actor: 'AGENT' | 'USER' | 'SAFETY_GUARD' | 'SYSTEM';
  action: string;
  is_irreversible: boolean;
  status: 'SUCCESS' | 'BLOCKED' | 'PENDING_APPROVAL' | 'APPROVED' | 'REJECTED' | 'KILLED';
  details: string;
  payload?: Record<string, any>;
  timestamp: string;
}

export interface AuditStats {
  total_actions: number;
  reversible_count: number;
  irreversible_count: number;
  human_interventions: number;
  blocked_actions: number;
  compliance_score_pct: number;
}
